base/collection_gocb.go

```diff
@@ -40,11 +40,23 @@ var (
 func AsCollection(dataStore DataStore) (*Collection, error) {
 	collection, ok := dataStore.(*Collection)
 	if !ok {
-		return nil, errors.New("dataStore is not a *Collection")
+		return nil, fmt.Errorf("dataStore is not a %T", dataStore)
```

Comment:
This change doesn't look correct? Maybe this:

```suggestion
		return nil, fmt.Errorf("dataStore is not a *Collection (got %T)", dataStore)
```